docs/pages/experiments/material-ui/v7-themes.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+import DefaultPropsProvider from '@mui/material/DefaultPropsProvider';
+import Button from '@mui/material/Button';
+import md2Slots from '@mui/material/md2';
+import md3Slots from '@mui/material/md3';
+import ShoppingCart from '@mui/icons-material/ShoppingCart';
+
+const theme = createTheme();
+
+export default function MD3() {
+  return (
+    <ThemeProvider theme={() => theme}>
+      <Box
+        sx={{
+          display: 'grid',
+          gridTemplateColumns: 'repeat(auto-fill, minmax(400px, 1fr))',
+          placeItems: 'center',
+          gap: 3,
+          p: 3,
+        }}
+      >
+        <div>
+          <Button startIcon={<ShoppingCart />} variant="elevated">
+            Button
+          </Button>
+        </div>
+        <div>
+          <DefaultPropsProvider value={md2Slots}>
+            <Button startIcon={<ShoppingCart />}>Button</Button>
+          </DefaultPropsProvider>
+        </div>
+        <div>
+          <DefaultPropsProvider value={md3Slots}>
```

Comment:
It reminds me of Chakra v2: https://github.com/chakra-ui/chakra-ui/blob/main/packages/react/src/theme/recipes/avatar.ts. And PrimeVue: https://github.com/primefaces/primevue/tree/master/packages/themes/src/presets.

As far as I know, it didn't work for them: https://github.com/chakra-ui/chakra-ui/issues/4975, https://github.com/chakra-ui/chakra-ui/issues/7383.

Reply:
I see, good point for the bundle size.